src/app/yourcity/yourcity.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { YourcityService } from "../service/yourcity.service";
+
+@Component({
+  selector: 'app-yourcity',
+  templateUrl: './yourcity.component.html',
+  styleUrls: ['./yourcity.component.css']
+})
+export class YourcityComponent {
+  nameCity: any;
+  temp: any;
+  weather: any;
+  timeOfDay: string = 'day';
+  BlackOrWhite: string = 'afterWhite';
+  constructor(private YourcityService: YourcityService) {
+  }
+  ngOnInit(){
+    this.getLocation();
+    this.SetBackgroundBasedOnTime();
+  }
+  getLocation() {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        const latitude = position.coords.latitude;
+        const longitude = position.coords.longitude;
+
+        this.YourcityService.getNameByCoords(latitude, longitude).subscribe((data) => {
```

Comment:
not the right way to use Observable, use pipe and `switchMap `operator for switching stream and receiving next request.